Automattic/wp-calypso
# Store Products: Switch fetchProducts to use the data layer

This switches over the `fetchProducts` action to call on the data-layer to make the API request, rather than making the request directly. My goal is to clean up the products & promotions states so that we can use [the appliesTo component](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/app/promotions/fields/promotion-applies-to-field/index.js) in the in-progress [ProductSearch component](https://github.com/Automattic/wp-calypso/pull/19191) for both sections, and rely only on products' state.

I've moved the code in the promotions handler that re-requested product pages using `offset` into the products handler, and removed most of the product-related code from promotions. I didn't touch the `getPromotionableProducts` selector, but I did add `getAllProducts`, which should do the same thing @coderkevin (so you can easily switch to that, then remove the product duplication in promos 🎉  )

**To test**

- Run the tests: `npm run test-client client/extensions/woocommerce`
- Click through store to check that all products pages are loading correctly
- Click through the promotions to make sure that all products load correctly there, too
- There should be no visual changes in this PR
